### ISSUE TITLE
vision_service

### ISSUE THREAD

**Issue by suyash-rgb** [suyash-rgb/AarogyaMitra#24]
The primary available datasets are below by platform and structure:

---

### 1. Kaggle & Open Research Repositories

* **RxHandBD / Doctor's Handwritten Prescription Dataset**
* **Size & Structure:** 5,578 standardized, cropped word images ($128 \times 128$) extracted from real clinical prescriptions with ground-truth CSV labels.
* **Content:** 1,559 unique clinical entries covering brand names, generic formulations, dosage metrics (e.g., *TDS*, *BD*, *SOS*), and administration routes.
* **Where to find:** Available on Kaggle (`abrahametry/rxhandbd-handwritten-word-image-dataset`) and Zenodo (DOI: `10.5281/zenodo.18478741`).

* **Doctor's Handwritten Prescription BD Dataset (Mamun et al.)**
* **Size & Structure:** 4,680 segmented handwritten drug names mapped across 78 frequent pharmaceutical classes.
* **Content:** Focuses on standard South Asian brands (e.g., *Montair*, *Sergel*, *Napa*, *Maxpro*, *Nexum*).
* **Where to find:** Kaggle (`mamun1113/doctors-handwritten-prescription-bd-dataset`).

* **OCR-Processed Handwritten Prescriptions**
* **Size & Structure:** End-to-end paired prescription images and machine-readable ground truth annotations.
* **Best Use:** Benchmarking end-to-end HTR (Handwritten Text Recognition) and evaluating sequence-to-sequence text reconstruction.
* **Where to find:** Kaggle (`nadaarfaoui/ocr-processed-handwritten-prescriptions`).

---

### 2. Hugging Face Datasets

* **`naazimsnh02/medocr-vision-dataset`**
* **Size & Tasks:** ~2,460 document images structured for Vision-Language Models and Document-AI pipelines.
* **Content:** High-resolution prescription sheets, medical lab reports (CBC, LFT, Lipid profiles), and doctor notes paired with full ground-truth transcriptions (doctor details, patient age, medication lists, dosages).
* **Best Use:** Document Visual Question Answering (DocVQA) and direct VLM instruction tuning.

* **`avi-kai/Medical_Prescription_Handwritten_Words`**
* **Size & Structure:** Image-to-text pairs of individual cursive physician tokens.
* **Best Use:** Training baseline vision encoders or fine-tuning transformer OCRs (e.g., TrOCR) on individual cursive ligatures.

---

### 3. Auxiliary Indian Pharmaceutical Lexicons (Essential for Grounding)

* **`junioralive/Indian-Medicine-Dataset` (GitHub)**
* A structured database (JSON/CSV) of over 10,000+ Indian commercial drug brands, manufacturer records, pack sizes, active salts, and standard strengths (e.g., *Augmentin 625 Duo*, *Dolo 650*, *Pantocid 40*).
* **Why it matters:** Vision models often misread single characters in handwriting (e.g., confusing an `o` for an `a`). Pairing this dataset with AarogyaMitra provides a fuzzy-matching lookup dictionary to validate extracted drug names against genuine market formulations.

---

### The Practical Limitation: Word-Level vs. Full-Page Data

Public datasets in this domain typically provide **cropped word images** rather than full-page sheets. This is driven by healthcare privacy laws (redaction of patient identifying data) and layout segmentation requirements.

Because of this, an end-to-end VLM fine-tuned on full unsegmented images can struggle with full-page spatial orientation. The most reliable architecture for this data is a hybrid workflow:

1. Use an object detector (YOLOv8 or PaddleOCR layout parser) to crop the prescription into medicine lines and Rx notes.
2. Pass those crops through a lightweight handwriting model trained on datasets like **RxHandBD**.
3. Feed the resulting raw text into your fine-tuned **AarogyaMitra Qwen model** to correct typos, infer dosage intervals, and produce structured triage notes.

**Comment by suyash-rgb** [comment]
To fine-tune a model like Qwen to accurately transcribe handwritten prescriptions, the required dataset size depends on the scope of deployment:

<html>
<body>
<html><head></head><body>
Deployment Scope | Labeled Image Volume | Writer Diversity | Expected Word Accuracy
-- | -- | -- | --
Proof of Concept / Single Clinic | 1,500 – 3,000 crops | 5 – 10 doctors | ~70 – 80% (fixed layout/handwriting)
Robust Regional Deployment | 15,000 – 30,000 full sheets | 150 – 300+ doctors | ~85 – 90% (diverse cursive styles)
Production-Grade (Commercial) | 100,000+ (Real + Synthetic) | 1,000+ doctors | >95% (open-vocabulary)

</body></html>
</body>
</html>
